tests/unit_tests/configuration/test_configurator.py

```diff
@@ -187,3 +181,49 @@ def test_initialize_output(configurator, args_dict):
     configurator.config["output_file"] = "unit_test.txt"
     configurator._initialize_output()
     assert configurator.config["output_file"] == "unit_test.txt"
+
+
+def test_fill_from_environmental_variables(configurator):
+    configurator.parser.initialize_output_arguments()
+    configurator._fill_from_command_line(arg_list=[])
+
+    _config_save = copy(configurator.config)
+
+    # this is not a configuration parameter and therefore should not be set
+    os.environ["TEST_ENV_VARIABLE"] = "test_value"
+    configurator._fill_from_environmental_variables()
+    if "TEST_ENV_VARIABLE" in os.environ:
+        del os.environ["TEST_ENV_VARIABLE"]
+    assert "test_env_variable" not in configurator.config
+
+    # this is a valid configuration parameter, but already configured
+    # _fill_from_environmental_variables() should not change it
+    os.environ["LOG_LEVEL"] = "DEBUG"
+    configurator._fill_from_environmental_variables()
+    assert configurator.config["log_level"] == _config_save["log_level"]
```

Comment:
Should this also compare to the actual log_level (e.g., `== INFO`) so that the test does not pass accidentally if the default configuration is "DEBUG"?

Reply:
The test depends on that the default value for log_level as defined in [here](https://github.com/gammasim/simtools/blob/e55809c6c90f3ed8b041d30ff9fd7542e84dab48/simtools/configuration/commandline_parser.py#L159). 

Do you see this an issue? Will we ever change the default level to "debug"? 

Note that we nowhere change the actually value of logging, so what I would have to add is a check if `config["log_level"] = "DEBUG"` , in this case use "INFO". Should I do this?

Reply:
This test, as it is, depends on a value set somewhere else. If, for whatever reason, you change the default value to "debug" (extremely unlikely, I know, but it doesn't matter), then this test would not cover this case anymore. Since the tests have to stand on their own and always test what they are meant to test, I suggest to change it to 
```
assert configurator.config["log_level"] == _config_save["log_level"] == "INFO"
```
(not that I am not sure if `"INFO"` is the exact representation, I did not check.)
This will make this test robust against changes somewhere else and would fail if you do change somewhere else and alert you to change also this test.

Reply:
Perfect, thanks!